gui/gui-manager.cpp

```diff
@@ -138,35 +138,37 @@ void GuiManager::initIconsSet() {
 		delete iconDir;
 	}
 
+	dat = nullptr;
+
 	const char fname[] = "gui-icons.dat";
-	Common::File *file = new Common::File;
 
 	if (ConfMan.hasKey("themepath")) {
 		Common::FSNode *fs = new Common::FSNode(normalizePath(ConfMan.get("themepath") + "/" + fname, '/'));
-
-		if (!fs->exists()) {
-			delete fs;
-		} else {
+		if (fs->exists()) {
 			dat = Common::makeZipArchive(*fs);
 		}
+		delete fs;
```

Comment:
I don't understand this change. This is handled internally when ZipFile is destroyed.

Reply:
No this is not because we are passing a reference to FSNode not a pointer to a stream.
The called prototype is this one: `Archive *makeZipArchive(const FSNode &node);` which does this:
```
return makeZipArchive(node.createReadStream());
```

Reply:
Thanks!